cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceService.java

```diff
@@ -25,21 +25,36 @@
 import com.agiletec.plugins.jacms.aps.system.services.resource.model.AbstractResource;
 import com.agiletec.plugins.jacms.aps.system.services.resource.model.ResourceDto;
 import com.agiletec.plugins.jacms.aps.system.services.resource.model.ResourceInterface;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
 import org.entando.entando.aps.system.exception.RestServerError;
+import org.entando.entando.aps.system.services.component.ComponentUsageEntity;
 import org.entando.entando.aps.system.services.DtoBuilder;
+import org.entando.entando.aps.system.services.component.IComponentDto;
+import org.entando.entando.aps.system.services.component.IComponentUsageService;
 import org.entando.entando.aps.system.services.IDtoBuilder;
 import org.entando.entando.aps.system.services.category.CategoryServiceUtilizer;
 import org.entando.entando.aps.system.services.group.GroupServiceUtilizer;
 import org.entando.entando.ent.util.EntLogging.EntLogger;
 import org.entando.entando.ent.util.EntLogging.EntLogFactory;
+import org.entando.entando.web.common.model.PagedMetadata;
+import org.entando.entando.web.common.model.RestListRequest;
+import org.springframework.beans.factory.annotation.Autowired;
 
 public class ResourceService implements IResourceService,
-        GroupServiceUtilizer<ResourceDto>, CategoryServiceUtilizer<ResourceDto> {
+        GroupServiceUtilizer<ResourceDto>, CategoryServiceUtilizer<ResourceDto>, IComponentUsageService {
 
     private final EntLogger logger = EntLogFactory.getSanitizedLogger(this.getClass());
 
     private IResourceManager resourceManager;
     private IDtoBuilder<ResourceInterface, ResourceDto> dtoBuilder;
+    
+    @Getter(AccessLevel.PROTECTED)@Setter
```

Comment:
I am probably missing some information and the full picture. This is just my opinion, but I think that since the 'resourceServiceUtilizers' variable is passed by composition it is better to inject it through the constructor into the derived classes. This way we can keep the 'resourceServiceUtilizers' variable completely private without exposing the setter and delegate the task of providing this value to the subclasses during construction.

Reply:
Fized